shared/common-adapters/button.desktop.js

```diff
@@ -94,29 +98,53 @@ export default class Button extends Component {
       }
     }
 
+    if (this.props.waiting) {
+      labelStyle = {
+        ...labelStyle,
+        opacity: 0
+      }
+    }
+
+    let outterStyle = {position: 'relative'}
+
     if (this.props.fullWidth) {
-      backgroundStyle = {
-        ...backgroundStyle,
         // Using minWidth here means we can't have a full-width button on the
         // same line/row as another button, the right thing is very unlikely to
         // happen.  The alternative is 'flex: 1' here, which would work but is
         // dangerous, because we'd be modifying our container.
         //
         // So let's just say that a fullWidth button can't have siblings.
-        minWidth: '100%'
-      }
+      outterStyle = {...outterStyle, minWidth: '100%'}
+      backgroundStyle = {...backgroundStyle, minWidth: '100%'}
+    }
+
+    if (this.props.waiting) {
+      outterStyle = {...outterStyle, cursor: 'wait'}
+    }
+
+    let label = this.props.label
+
+    if (this.props.more) {
+      label = '•••'
     }
 
     return (
-      <FlatButton
-        onClick={this.props.onClick}
-        style={{...backgroundStyle, ...smallStyle, ...this.props.style}}
-        labelStyle={{...styles2.buttonLabel, ...labelStyle}}
-        label={this.props.label || this.props.more && '•••'}
-        primary={this.props.type === 'Primary'}
-        secondary={this.props.type === 'Secondary'}
-        disabled={this.props.disabled}
-      />
+      <div style={outterStyle}>
```

Comment:
we wrap the button. the button itself gets opacity changes when disabled so let's not have our overlay get that too